test/rekt/features/channel/data_plane.go

```diff
@@ -265,3 +289,96 @@ func addControlPlaneDelivery(fs *feature.FeatureSet) {
 		fs.Features = append(fs.Features, f)
 	}
 }
+
+func channelAcceptsBinaryContentMode(ctx context.Context, t feature.T) {
+	channelName := "mychannelimpl"
+	contenttypes := []string{
+		"application/vnd.apache.thrift.binary",
+		"application/xml",
+		"application/json",
+	}
+	for _, contenttype := range contenttypes {
+		source := feature.MakeRandomK8sName("source")
+		eventshub.Install(source,
+			eventshub.StartSenderToResource(channel_impl.GVR(), channelName),
+			eventshub.InputHeader("ce-specversion", "1.0"),
+			eventshub.InputHeader("ce-type", "sometype"),
+			eventshub.InputHeader("ce-source", "200.request.sender.test.knative.dev"),
+			eventshub.InputHeader("ce-id", uuid.New().String()),
+			eventshub.InputHeader("content-type", contenttype),
+			eventshub.InputBody("{}"),
+			eventshub.InputMethod("POST"),
+		)(ctx, t)
+
+		store := eventshub.StoreFromContext(ctx, source)
+		events := knconf.Correlate(store.AssertAtLeast(ctx, t, 2, knconf.SentEventMatcher("")))
+		for _, e := range events {
+			if e.Response.StatusCode < 200 || e.Response.StatusCode > 299 {
+				t.Errorf("Expected statuscode 2XX for sequence %d got %d", e.Response.Sequence, e.Response.StatusCode)
+			}
+		}
+	}
+}
+
+func channelAcceptsStructuredContentMode(ctx context.Context, t feature.T) {
+	channelName := "mychannelimpl"
+	contenttype := "application/cloudevents+json"
+	bodycontent := `{
+    "specversion" : "1.0",
+    "type" : "sometype",
+    "source" : "json.request.sender.test.knative.dev",
+    "id" : "2222-4444-6666",
+    "time" : "2020-07-06T09:23:12Z",
+    "datacontenttype" : "application/json",
+    "data" : {
+        "message" : "helloworld"
+    }
+}`
+	source := feature.MakeRandomK8sName("source")
+	eventshub.Install(source,
+		eventshub.StartSenderToResource(channel_impl.GVR(), channelName),
+		eventshub.InputHeader("content-type", contenttype),
+		eventshub.InputBody(bodycontent),
+		eventshub.InputMethod("POST"),
+	)(ctx, t)
+
+	store := eventshub.StoreFromContext(ctx, source)
```

Comment:
here, I would make a receiver and subscribe it to the channel and then create the store on that.

Right now, you are looking at the events being stored at the source, rather than after the channel passed the event along

Reply:
Took inspiration from ```brokerAcceptsStructuredContentMode``` from ```test/rekt/features/broker/data_plane.go``` would be great if you have look at it

Reply:
Also it seems store is created by the test framework during install of source so just had a thought we might avoid creating stores ourselves ?

Reply:
Yeah that test is correct there, because that is only verifying that the broker accepts the event. However, this is trying to test something different so we can't just copy the broker test directly. Here we want to verify that the channel does not modify the event as it passes to a subscriber, hence we need to create the store on the subscriber and see if the schema version is correct.

In terms of creating the store, this method is just fetching the store from the context so that we can interact with it. But yes, you can also use the `OnStore` method directly (I personally do that a lot of the time). See code here: https://github.com/knative/eventing/blob/7d350ea0c6be334ec285a2d8f5dfb660682e45f7/test/rekt/features/broker/source_to_sink.go#L136-L142

The key is that you need to check the event on the subscriber, not the channel. Otherwise you aren't correctly testing the behaviour. Hope that makes sense!

Reply:
Whoops, looking at this again and I put this comment on the wrong line where you add get a store - sorry! This should be done on the next store for the next function, ShouldNotUpdateVersion